### PR TITLE
Add test for our container definitions

### PR DESCRIPTION
**Pull request by okurz**
This adds a simple test script for our containers enabled in a separate
github actions job definition. Keep in mind that the containers rely on
the published openSUSE packages on download.opensuse.org and not the
code within this repository in the current state with the exception of
the container build files themselves, e.g. Dockerfiles.

Related progress issue: https://progress.opensuse.org/issues/81118

### CODE REVIEW

**Comment by okurz** [comment]
@perlpunk if I understood https://github.com/perlpunk/os-autoinst-common/blob/readme/README.md#make-changes-in-host-repo-first-and-push-after-approval correctly then after merging this PR into os-autoinst I can still push the changes from https://github.com/os-autoinst/os-autoinst/pull/1602/commits/d8aa56fd96823a75cbda2643663eb84b950f5384 upstream to os-autoinst-common, is this correct?

**Comment by perlpunk** [comment]
> @perlpunk if I understood https://github.com/perlpunk/os-autoinst-common/blob/readme/README.md#make-changes-in-host-repo-first-and-push-after-approval correctly then after merging this PR into os-autoinst I can still push the changes from [d8aa56f](https://github.com/os-autoinst/os-autoinst/commit/d8aa56fd96823a75cbda2643663eb84b950f5384) upstream to os-autoinst-common, is this correct?

Yes, this should be possible

**Comment by okurz** [comment]
PR is all ready and hoping for some review :)

File: tools/static_check_containers [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# use github internal registry if available, e.g. within github actions
+if [ "$(curl -s -w "%{http_code}" https://ghcr.io/v2/ -o /dev/null)" = "401" ]; then
+    img=hadolint/hadolint
+else
+    img=ghcr.io/hadolint/hadolint
+fi
+for i in container/*/Dockerfile*; do
+    echo "# Static check of $i"
+    docker run --rm -i $img < $i
```
Comment (kalikiana) [inline tools/static_check_containers:13]: I didn't know [hadolint](https://hadolint.github.io/hadolint/) before. The logo is a cat looking embarassed whilst doing its business. It must be good 😁 I like it

**Comment by okurz** [comment]
I assume this means you approve :)

**Review by kalikiana — APPROVED** [review]

**Pull request merged**